### PR TITLE
feat(ios): textarea contentHeight fix with new return property textareaHeight in "change" event

### LINKED ISSUE #13413: iOS: textarea missing correct contentHeight on content change - event "change" now with correct height (no postlayout needed)
### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

Textarea on iOS missing a correct contentHeight feature.

### Solution

I fixed this on event "change" for textarea which will always return the correct textarea height when the content of textarea is changing.

```
textarea.addEventListener('chage',function(e) {
        console.log(e.textareaHeight);
        console.log(e.value);
});

```

### Alternatives

_No response_

### Platforms

iOS

### PR DESCRIPTION
**Pull request by mbender74**
Fixes #13413

### CODE REVIEW

File: apidoc/Titanium/UI/TextArea.yml [hunk 1]
```diff
@@ -518,6 +518,11 @@ events:
       - name: value
         summary: New value of this text area.
         type: String
+      - name: textareaHeight
```
Comment (hansemannn) [inline apidoc/Titanium/UI/TextArea.yml:521]: Needs to be `textAreaHeight`

File: iphone/Classes/TiUITextField.m [hunk 2]
```diff
@@ -564,7 +564,7 @@ - (BOOL)textFieldShouldEndEditing:(UITextField *)tf
 - (BOOL)textFieldShouldClear:(UITextField *)tf
 {
   // we notify proxy so he can serialize in the model
-  [(TiUITextFieldProxy *)self.proxy noteValueChange:@""];
+  [(TiUITextFieldProxy *)self.proxy noteValueChange:@"":nil];
```
Comment (hansemannn) [inline iphone/Classes/TiUITextField.m:567]: Why not use the named parameter?

File: iphone/Classes/TiUITextWidgetProxy.m [hunk 3]
```diff
@@ -118,13 +118,22 @@ - (BOOL)focused:(id)unused
...
-- (void)noteValueChange:(NSString *)newValue
+- (void)noteValueChange:(NSString *)newValue:(NSNumber *)contentHeight
 {
   NSString *oldValue = [TiUtils stringValue:[self valueForKey:@"value"]];
   if (![oldValue isEqual:newValue]) {
     [self replaceValue:newValue forKey:@"value" notification:NO];
+
+    NSMutableDictionary *event = [NSMutableDictionary dictionary];
+    if (contentHeight != nil) {
+      [event setValue:contentHeight forKey:@"textareaHeight"];
```
Comment (hansemannn) [inline iphone/Classes/TiUITextWidgetProxy.m:129]: textAreaHeight as well. Or just `height` since it's in the implicit context

**Review by hansemannn — CHANGES_REQUESTED** [review]
The backwards compatibility looks fine, but please see the changes before merging. Thx!

File: iphone/Classes/TiUITextField.m (hunk 2 above)
Comment (mbender74) [reply iphone/Classes/TiUITextField.m:567]: What named parameter do you mean?

**Review by mbender74 — COMMENTED** [review]
changes made
- event property is now "contentHeight" which is more self explaining and conform with similar properties for other views. "height" would not make sense, because, the textArea has a own height, and this event property reflects the inner scrollView height of the text widget.
- API docs updated to that

**Review by hansemannn — APPROVED** [review]
Looks solid, let's go! Thanks for this! 🚀

**Pull request merged**